dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/UnitOfWorkAwareProxyFactory.java

```diff
@@ -100,8 +100,9 @@ public UnitOfWorkAwareProxyFactory(HibernateBundle<?>... bundles) {
 
     /**
      * @return a new
+     * @param sessionFactories
      */
-    public UnitOfWorkAspect newAspect() {
-        return new UnitOfWorkAspect(sessionFactories);
+    public UnitOfWorkAspect newAspect(ImmutableMap<String, SessionFactory> sessionFactories) {
+        return new UnitOfWorkAspect(this.sessionFactories);
```

Comment:
Shouldn't it be `sessionFactories` instead of `this.sessionFactories`? If yes, would it be possible to overload `newAspect` instead of changing its signature?